src/core/dbus-execute.c

```diff
@@ -1084,6 +1085,25 @@ static int property_get_protect_hostname(
         return sd_bus_message_append_basic(reply, 'b', &b);
 }
 
+static int property_get_protect_hostname_ex(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *reply,
+                void *userdata,
+                sd_bus_error *error) {
+
+        ExecContext *c = ASSERT_PTR(userdata);
+        _cleanup_free_ char *s = NULL;
+
+        s = strjoin(protect_hostname_to_string(c->protect_hostname), c->private_hostname ? ":" : NULL, c->private_hostname);
+        if (!s)
+                return -ENOMEM;
```

Comment:
Hmm, is this really the right approach? When exposing properties we usually try to hand out structured data, i.e. change the dbus signature of `ProtectHostname` to `ss` instead? It's added only recently and hence can still be modified.